src/tracking.ts

```diff
@@ -0,0 +1,222 @@
+import { getItem, RequestError, setItem } from ".";
+
+export const STORAGE_KEY = "searchio_events";
+const EXPIRY_DAYS = 30;
+const FUNNEL_ENTRY_TYPES = ["click"];
+
+type Metadata = Record<string, boolean | number | string>;
+interface EventState {
+  queryId: string;
+  type: string;
+  timestamp: number;
+  submitted: boolean;
+  metadata?: Metadata;
+}
+type Value = string | number;
+type Identifier = "result_id" | "redirect_id" | "promotion_id";
+
+export class SearchIOAnalytics {
+  account: string;
+  collection: string;
+  endpoint: string;
+  events: Record<Value, EventState[]>;
+  queryId?: string;
+
+  constructor(
+    account: string,
+    collection: string,
+    endpoint: string = "https://api.search.io"
```

Comment:
Wondering if it's possible to just take a `Client` as the single operand here.

Reply:
I don't think it's worth introducing a dependency on `Client` purely for the `account` and `collection` values.